test-framework/src/test/java/org/apache/brooklyn/test/framework/TestSshCommandIntegrationTest.java

```diff
@@ -58,12 +62,39 @@
     @Override
     public void setUp() throws Exception {
         super.setUp();
-
         testEntity = app.createAndManageChild(EntitySpec.create(TestEntity.class)
                 .location(TestApplication.LOCALHOST_MACHINE_SPEC));
     }
 
     @Test(groups = "Integration")
+    public void shouldRetryCommandWithinTimeout() throws Exception {
+        TestSshCommand testWithCmd = app.createAndManageChild(EntitySpec.create(TestSshCommand.class)
+                .configure(TARGET_ENTITY, testEntity)
+                .configure(COMMAND, "_djaf-_f£39r24")
+                .configure(RUN_DIR, "/tmp")
+                .configure(TIMEOUT, Duration.TEN_SECONDS)
+                .configure(ASSERT_STATUS, makeAssertions(ImmutableMap.of(EQUALS, 0))));
+
+        // Run the test's assertions in parallel with the app.
+        Thread t = new Thread() {
+            @Override
+            public void run() {
+                app.start(ImmutableList.<Location>of());
+            }
+        };
+        t.start();
+
+        // Times out in one second.
+        EntityAsserts.assertAttributeContinuallyNotEqualTo(testWithCmd, Attributes.SERVICE_STATE_ACTUAL, Lifecycle.RUNNING);
+
+        // Replace command with a valid one and the test should start to pass.
+        testWithCmd.config().set(COMMAND, "true");
+        EntityAsserts.assertAttributeEqualsEventually(testWithCmd, Attributes.SERVICE_STATE_ACTUAL, Lifecycle.RUNNING);
+
+        t.join();
```

Comment:
I'd prefer a timeout in the join - so if we introduce something that causes the thread to not return, then the test would fail rather than hanging. But no strong feelings (given we call app.start all over the place in the main thread for tests).